### DISCUSSION TITLE
PPL taking stuff from my backpack

### DISCUSSION THREAD

**Discussion by Kinghans1231** [BeansGalaxy/Beans-Backpacks-2#123] [Feedback] [answered]
Im not sure if this exists but it would be cool if there was an option to "lock" the backpack to prevent people taking my stuff inside even when its on my back. Thoughts?

**Comment by Erinbjorn** [comment, marked as answer]
Isn't that what the first gamerule in the config does?

"lockBackpackNotOwner" - Lock placed Backpacks if a player did not place it

**Reply by Kinghans1231** [reply]
Oh, im not quite familiar with config files and such, does that setting also affect worn backpacks?

**Reply by Erinbjorn** [reply]
I have no idea as I dont have another player to test this, but according to the description it would seem to do what you're asking for here

**Reply by BeansGalaxy** [reply]
if you're OP in the world you can use the command `/beansmod gamerule lockBackpackNotOwner true` to lock backpacks if you're not the owner, BUT there's no way to lock it on your back. I have some features I could add to help stop this though

1. Locks - Not very vanilla feeling but solves the problem. I've been playing with this idea for a while but not 100% on the *feel* of this mechanic.
2. In vanilla, if a menu closes and you have an item in your cursor it'll automatically jump to your inventory if you have a free slot. Since moving a block and turning around closes the menu, it would make since if the cursor item defaults to the backpack's inventory.
3. If you're not holding your hotkey then no one can open your equipped backpack. I think this is best as another gamerule since I think it would get annoying on casual servers. 

Lmk if you have any ideas for these changes and if they would actually help :) thanks for the feedback.

**Reply by Kinghans1231** [reply]
That last one seems great! But for some reason hard to do no? Loving the mod btw

**Reply by BeansGalaxy** [reply]
The hard part is syncing your CTRL key with the server but I already did that since behind the scenes the placing mechanic is pretty complex. Since I have everything on the server already that one's actually the easiest

**Reply by Kinghans1231** [reply]
Looking forward to that feature.

**Reply by BeansGalaxy** [reply]
Update 20.1-0.25-v2 on Modrinth has `/beansmod gamerule lockBackpackNoKey`. This'll require the player to hold their hotkey down for other player to access their backpack. In the next update though I'm gonna remove this and let the player Shift + RClick the backpack item in their hand to lock it. Literally posted the update and thought of it.

**Comment by BeansGalaxy** [comment]
Backpacks can now be locked by holding the backpack item in your hand and shift right clicking it as of 20.1-0.26-v2. `/beansmod gamerule lockBackpackNoKey` is still in the game but has no use since you can now lock backpacks on an individual basis. :)

**Reply by VideoGames1000** [reply]
This seems like something you should really have on the modrinth page, since until I went to check this (intending to suggest this very feature) I was originally turned off from the idea of using the mod. Now that I know about this though, it's a far more tempting backpack solution going forward.
